apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/_components/claim-conditions/claim-conditions-form/hooks.ts

```diff
@@ -114,8 +128,18 @@ export function setClaimPhasesTx(
   const phases = rawPhases.map((phase) => {
     return {
       startTime: toDate(phase.startTime),
-      maxClaimableSupply: toBigInt(phase.maxClaimableSupply),
-      maxClaimablePerWallet: toBigInt(phase.maxClaimablePerWallet),
+      // The input from client-side is non-wei, but the extension is expecting value in wei
+      // so we need to convert it using toUnits
+      maxClaimableSupply: toBigInt(
+        phase.maxClaimableSupply,
+        baseOptions.type === "erc20" ? baseOptions.decimals : undefined,
```

Comment:
feels a bit confusing to bake this logic inside toBigInt - i think i would separate it at this level, and do an explicit to toUnits here instead

Reply:
Ok

Reply:
Updated ^